codalab/worker/worker_run_state.py

```diff
@@ -164,6 +164,17 @@ def _transition_from_PREPARING(self, run_state):
         if run_state.is_killed:
             return run_state._replace(stage=RunStage.CLEANING_UP)
 
+        # Check CPU and GPU availability
+        try:
+            cpuset, gpuset = self.assign_cpu_and_gpu_sets_fn(
+                run_state.resources.cpus, run_state.resources.gpus
+            )
+        except Exception as e:
+            message = "Cannot assign enough resources: %s" % str(e)
```

Comment:
This really shouldn't happen - the bundle manager would have screwed up if we got here...  Make the message clear that this is unexpected.  Do we see this message in practice?

Reply:
Can you update the `message`?  "... %s (is happening due to bugs or errors in the bundle manager assignment)"

Reply:
I should have updated this yesterday. One thing that I just realized that this message will show up on the web UI on the bundle page. Do we want to mention the reason there? Or we log this message separately in the host machine?

Reply:
Ah, since it's going to show up in the UI, then we can maybe make it a bit more friendly, like "Unexpectedly unable to assign ..."